### PR TITLE
bulletproof termination after set amount of time [ of inactivity ]

### PR DESCRIPTION
**Pull request by m4444x**
Implemented a new method to make sure that the program is terminated after a set amount of time has passed. It works by setting a timer via alert() and it should be bulletproof under all conditions.

Also adds the ability to choose between continuous and non-continuous scanning. In non-continuous mode it will terminate when the time is up while in continuous mode it will reset the timer every time a beacon is detected. The timer can also be disabled by setting the timeout <= 0.

### CODE REVIEW

File: scanner.c [hunk 1]
```diff
@@ -113,35 +117,53 @@ int main()
...
 					info = (le_advertising_info *)offset;
 					char addr[18];
-					ba2str(&(info->bdaddr), addr);
-                    printf("%s %d", addr, (int8_t)info->data[info->length]);
-                    for (int i = 0; i < info->length; i++) {
-                       printf(" %02X", (unsigned char)info->data[i]);
-                    }
-                    printf("\n");
+					ba2str( &(info->bdaddr), addr);
+					if ( info->data[0] == 0x1e && info->data[1] == 0xff )
```
Comment (davidgyoung) [inline scanner.c]: Why add this check?  The is will restrict detections to only certain types of advertisements, which would make the tool less flexible.

**Review by davidgyoung — CHANGES_REQUESTED** [review]
Looks great, @m444x, thanks for the changes.  

I have one question about an if condition.  Please take a look.

File: scanner.c (hunk 1 above)
Comment (m4444x) [reply scanner.c]: Good question. It's probably a left over from my attempts to decode/filter some messages. Let me fix that.

Comment (m4444x) [reply scanner.c]: OK done. See below

**Comment by m4444x** [comment]
Added a few things to mimic the old behaviour ( exit code 0, 10 seconds timeout ).
So it should be a drop in replacement for people using the original program in their scripts.

**Pull request merged**